src/Files.App/Utils/Storage/Helpers/FileThumbnailHelper.cs

```diff
@@ -17,6 +18,22 @@ public static class FileThumbnailHelper
 			return await Win32Helper.StartSTATask(() => Win32Helper.GetIcon(path, (int)size, isFolder, iconOptions));
 		}
 
+		/// <summary>
+		/// Returns thumbnail for given file or folder using Storage API
+		/// </summary>
+		public static async Task<byte[]?> GetIconAsync(IStorageItem item, uint requestedSize, ThumbnailMode thumbnailMode, ThumbnailOptions thumbnailOptions)
```

Comment:
Is thumbnail mode always single?

Reply:
Both ListView and SingleItem are used

Reply:
I think we can reduce complexity by just using SingleItem.

Reply:
Done!

Reply:
Thanks!